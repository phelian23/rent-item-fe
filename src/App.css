@media all and (max-width: 600px) {
  .sidebar,
  .overlay {
    position: fixed;
    top: 0;
    bottom: 0;
  }

  .sidebar {
    left: 0;
    min-width: 200px;
    max-width: 350px;
    overflow-x: hidden;
    background-color: #f8f4f4;
    transition: all ease 0.4s;
    flex-direction: column;
    justify-content: flex-start !important;
    align-items: flex-start !important;
    padding-top: 10%;
    padding-left: 5% !important;
  }

  .nav-cus-item,
  .signout {
    margin: 10% 0;
  }

  .nav-logo-cont {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    font-weight: bold;
    color: #464747 !important;
  }

  .nav-logo-cont img {
    width: 50%;
    height: auto;
    border-radius: 50%;
    margin-bottom: 5%;
  }

  .overlay {
    position: fixed;
    left: 0;
    top: 0;
    right: 0;
    z-index: 30;
    background-color: #fff;
    box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.4);
    transition: all ease 0.4s;
    overflow: hidden;
  }

  .main-navigation.active {
    left: 60%;
    top: 5%;
    bottom: 3%;
  }

  .add-btn {
    width: 100%;
    position: absolute;
    bottom: 0;
    border-radius: 0 !important;
    background-color: tomato !important;
    border: tomato !important;
    min-width: 250px;
  }

  .auth-con {
    position: fixed;
    top: 0;
  }

  .auth-icon {
    position: fixed;
    left: 7%;
    top: 7%;
    font-size: 30px;
    color: #161616;
  }

  .card-text {
    font-size: 12px;
  }
}

.overlay {
  color: #464747;
}

.links {
  text-decoration: none;
  color: #000;
}

.auth-con {
  width: 100%;
  height: 100%;
  background:
    url(https://thumbs.dreamstime.com/b/background-grey-brick-wall-great-cool-background-wallpaper-background-grey-brick-wall-great-cool-171615280.jpg)
    center center fixed;
}

.auth-form {
  width: 100%;
  height: 100%;
  background: rgba(230, 222, 222, 0.6);
}

.auth-cont {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 95vh;
  text-align: center;
}

.form-input1 {
  display: block;
  width: 100%;
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #212325;
  background-color: rgb(255, 255, 255, 0.5);
  background-clip: padding-box;
  border: 1px solid #ced4da;
  appearance: none;
  border-radius: 50px;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.form-input2 {
  border-radius: 1rem;
}

.scoot-list-cont {
  height: 100vh;
}

.scoot-card {
  position: relative;
  display: flex;
  flex-direction: column;
  width: 90vw;
  max-width: 350px;
  height: 70vh;
  background-color: #fff;
  background-clip: border-box;
  border: 0;
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15);
}

.details-card {
  position: relative;
  display: flex;
  flex-direction: column;
  width: 100vw;
  max-width: 375px;
  height: 50vh;
  background-clip: border-box;
  border: 0;
}

.img-text {
  position: absolute;
  bottom: 2%;
  width: 100%;
  color: #e95940;
}

.list-wrapper {
  overflow: hidden;
}

.list-group-item {
  border: 0 !important;
}

.list-con,
.tab-container {
  overflow-x: auto;
  overflow-y: hidden;
}

.list-con {
  margin: 0;
  padding: 0;
}

.form-btn {
  color: #fff !important;
  background-color: #fa5133 !important;
  border-color: #e95940 !important;
  border-radius: 50px !important;
  box-shadow: 0 0.5rem 1rem rgba(233, 89, 64, 0.4) !important;
}

.menu-icon,
.header-det {
  min-width: 270px;
}

.card-img-top {
  height: 70%;
}

.card-img-det {
  height: 100%;
}
