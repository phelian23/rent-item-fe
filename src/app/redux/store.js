import { configureStore } from '@reduxjs/toolkit';
import { combineReducers, compose, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import logger from 'redux-logger';
import itemReducer from './reducers/itemReducer';
import userReducer from './reducers/userReducer';
import favouriteReducer from './reducers/favouriteReducer';

const reducer = combineReducers({
  items: itemReducer,
  user: userReducer,
  favourites: favouriteReducer,

  // Add other reducers here
});

const composed = compose(applyMiddleware(thunkMiddleware, logger));

const store = configureStore({ reducer, composed });

export default store;
