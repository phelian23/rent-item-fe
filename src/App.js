import React from 'react';
import './App.css';
import NavBar from './app/components/Navbar';

function App() {
  return (
    <div>
      <NavBar />
    </div>
  );
}

export default App;
