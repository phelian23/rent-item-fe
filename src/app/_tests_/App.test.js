import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import App from '../../App';

describe('App', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <App />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
