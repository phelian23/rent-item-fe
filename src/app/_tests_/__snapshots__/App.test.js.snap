// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div>
  <div
    className="p-0"
  >
    <div>
      <nav
        className="d-block fixed-lg-top navbar navbar-expand-md navbar-light bg-light"
      >
        <div
          className="d-lg-flex sidebar container"
          id="responsive-navbar-nav"
        >
          <a
            className="mr-auto mr-lg-3 ml-3 ml-lg-0 nav-logo-cont navbar-brand"
            href="/"
          >
            <img
              alt=""
              className="d-inline-block align-top"
              height="40px"
              src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQzGIxlIopL2Rkz2W31izyQcASgFiW-bAHDEw&usqp=CAU"
              width="40px"
            />
             
            Scooter
          </a>
          <div
            className="navbar-nav"
            onKeyDown={[Function]}
          >
            <a
              className="nav-link"
              data-rr-ui-event-key="/signin"
              disabled={false}
              href="/signin"
              onClick={[Function]}
            >
              Login
            </a>
            <a
              className="nav-link"
              data-rr-ui-event-key="/signup"
              disabled={false}
              href="/signup"
              onClick={[Function]}
            >
              Sign Up
            </a>
          </div>
        </div>
      </nav>
    </div>
    <div
      className="main-navigation overlay"
    >
      <div
        className="tab-container d-flex justify-content-between align-item-center p-3"
      >
        <div
          className="menu-icon d-lg-none"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </div>
        <div
          className="d-lg-none"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 18a7.952 7.952 0 0 0 4.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0 0 18 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
            />
          </svg>
        </div>
      </div>
      <div
        className="scoot-list-cont"
      >
        <div
          className="text-center p-lg-3 header-det"
        >
          <h1>
            Scooter List
          </h1>
        </div>
        <div
          className="list-wrapper"
        >
          <ul
            className="d-flex list-con"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
